cmd/routedns/resolver.go

```diff
@@ -55,6 +64,9 @@ func instantiateResolver(id string, r resolver, resolvers map[string]rdns.Resolv
 			return err
 		}
 	case "doh":
+		r.Address = rdns.AddressWithDefault(r.Address, rdns.DoHPort)
+		r.BootstrapAddr = rdns.AddressWithDefault(r.BootstrapAddr, rdns.DoHPort)
+
```

Comment:
Not sure this is the right way to do it. The addresses can be URLs I think, and BootstrapAddr isn't (and shouldn't) be set for everything like this. I'll come back to this later

Reply:
I thought about that after I published it. Maybe we create an issue for it and push it for now knowning it can be a bug

Reply:
I wonder if it's sufficient for the function to return a blank string if the address is empty. Like

```go
if addr == "" {
  return addr
}
```

That would preserve blank values which are the default

Reply:
I like that idea!

Reply:
To improve on this do you think it would be better if it was just the host part of SplitHostPort because of the `:853` etc